test/core/xds/xds_certificate_provider_test.cc

```diff
@@ -36,15 +39,57 @@ namespace grpc_core {
 namespace testing {
 namespace {
 
-constexpr const char* kRootCert1 = "root_cert_1_contents";
-constexpr const char* kRootCert2 = "root_cert_2_contents";
+constexpr const char* kRootCert1Contents = "root_cert_1_contents";
+constexpr const char* kRootCert2Contents = "root_cert_2_contents";
 constexpr const char* kIdentityCert1PrivateKey = "identity_private_key_1";
 constexpr const char* kIdentityCert1 = "identity_cert_1_contents";
 constexpr const char* kIdentityCert2PrivateKey = "identity_private_key_2";
 constexpr const char* kIdentityCert2 = "identity_cert_2_contents";
 constexpr const char* kRootErrorMessage = "root_error_message";
 constexpr const char* kIdentityErrorMessage = "identity_error_message";
 
+constexpr absl::string_view kGoodSpiffeBundleMapPath =
+    "test/core/credentials/transport/tls/test_data/spiffe/"
+    "client_spiffebundle.json";
+
+constexpr absl::string_view kGoodSpiffeBundleMapPath2 =
+    "test/core/credentials/transport/tls/test_data/spiffe/"
+    "test_bundles/spiffebundle2.json";
+
+std::shared_ptr<RootCertInfo> kRootCert1() {
+  static const absl::NoDestructor<std::shared_ptr<RootCertInfo>> kRootCert1(
+      [] { return std::make_shared<RootCertInfo>(kRootCert1Contents); }());
```

Comment:
Why is this a lambda?  Can you're just directly instantiate it with `std::make_shared<RootCertInfo>(kRootCert1Contents)`?

Same thing below.

Reply:
I'm fine with not allocating it over and over; I was just saying that the lambda wasn't needed.  You can say something like this:

```
static const absl::NoDestructor<std::shared_ptr<RootCertInfo>> kRootCert1(
    std::make_shared<RootCertInfo>(kRootCert1Contents));
```

But the way you have it now is fine too.